### PR TITLE
feature(fill_db_data): log every step in between updates

### PR DESCRIPTION
**Pull request by dkropachev**
https://github.com/scylladb/scylla-cluster-tests/issues/3721
https://trello.com/c/kJCdDZb7/3598-make-upgrade-tests-report-in-details-step-it-done-and-time-it-took

Upgrade tests time to time is running slow
It is hard to identify where exactly delay is
Let's log everything between upgrades to see what is running slow

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] ~~New configuration option are added and documented (in `sdcm/sct_config.py`)~~
- [ ] ~~I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)~~
- [x] All new and existing unit tests passed (CI)
- [ ] ~~I have updated the Readme/doc folder accordingly (if needed)~~

### CODE REVIEW

File: sdcm/fill_db_data.py [hunk 1]
```diff
@@ -2976,6 +2983,7 @@ def _get_table_name_from_query(query):
         match = regexp.search(query)
         if match:
             return match.groupdict()["table_name"]
+        return None
```
Comment (vponomaryov) [inline sdcm/fill_db_data.py:2986]: Returning "None" is default behavior. Why do you add this line?

Comment (dkropachev) [reply sdcm/fill_db_data.py:2986]: it is fixing pylint issue, moved to separate commit

File: sdcm/fill_db_data.py [hunk 2]
```diff
@@ -3086,6 +3098,7 @@ def get_scylla_version(self):
         # NOTE: node.get_scylla_version() returns following structure of a scylla version:
         #       4.4.1-0.20210406.00da6b5e9
         #       And 'parse_version' behaves differently for full and short scylla versions
+        #       And 'parse_version' behaves differently for full and short scylla versions
```
Comment (vponomaryov) [inline sdcm/fill_db_data.py:3101]: Looks like mistakenly added copy of above line.

**Review by vponomaryov — CHANGES_REQUESTED** [review]
Small thing to fix, everything else LGTM

File: sdcm/fill_db_data.py (hunk 2 above)
Comment (dkropachev) [reply sdcm/fill_db_data.py]: Thanks, removed

**Review by vponomaryov — APPROVED** [review]

File: sdcm/fill_db_data.py [hunk 3]
```diff
@@ -3137,16 +3149,27 @@ def truncate_table(self, session, truncate):  # pylint: disable=no-self-use
         LOGGER.debug(truncate)
         session.execute(truncate)
 
+    @contextlib.contextmanager
+    def _execute_and_log(self, message):
```
Comment (roydahan) [inline sdcm/fill_db_data.py:3153]: The usage of context manager is nice, but is it really needed here?
Wouldn't just wrapping the calls with a "self.log.info"  and "timeit" been easier?

**Review by roydahan — APPROVED** [review]
Overall looks good and great initiative!

**Comment by fgelcer** [comment]
@roydahan @dkropachev , why wasn't this PR back ported to branch-4.5???
it makes debugging 4.5 so harder...

**Comment by roydahan** [comment]
Now it is backported if it helps.

**Pull request merged**